rp2040-hal/examples/rtc_sleep_example.rs

```diff
@@ -0,0 +1,169 @@
+
+    // Configure the clocks
+    let mut clocks = hal::clocks::ClocksManager::new(pac.CLOCKS);
+    // First, enable and wait for xosc to be stable.
+    let xosc = hal::xosc::setup_xosc_blocking(pac.XOSC, XTAL_FREQ_HZ).unwrap();
+
+    // use xosc at 12MHz for clk_ref -> clk_sys -> clk_peri
+    clocks
+        .reference_clock
+        .configure_clock(&xosc, XTAL_FREQ_HZ)
+        .unwrap();
+    clocks
+        .system_clock
+        .configure_clock(&clocks.reference_clock, XTAL_FREQ_HZ)
+        .unwrap();
+    clocks
+        .peripheral_clock
+        .configure_clock(&clocks.system_clock, XTAL_FREQ_HZ)
+        .unwrap();
+    // use xosc at 12MHz/256 for clk_rtc
+    clocks.rtc_clock.configure_clock(&xosc, 46875.Hz()).unwrap();
+    // Only leave the rtc's clock enabled while in deep sleep.
+    clocks.configure_sleep_en(
+        |_, w| unsafe {
+            // SAFETY: set bits as a shortcut to clear all fields to 0.
+            w.bits(0).clk_rtc_rtc().set_bit()
+        },
+        |_, w| w,
+    );
+
+    // The single-cycle I/O block controls our GPIO pins
+    let sio = hal::Sio::new(pac.SIO);
+
+    // Set the pins to their default state
+    let pins = gpio::Pins::new(
+        pac.IO_BANK0,
+        pac.PADS_BANK0,
+        sio.gpio_bank0,
+        &mut pac.RESETS,
+    );
+
+    // Configure GPIO 25 as an output to drive our LED.
+    let mut led = pins.gpio25.into_push_pull_output();
+
+    // Prepare the RTC for the example using the 1/1/0 (Day/Month/Year) at 0:00:00 as the initial
+    // day and time (it may not have been a Monday but it doesn't matter for this example.).
+    let mut rtc = hal::rtc::RealTimeClock::new(
+        pac.RTC,
+        clocks.rtc_clock,
+        &mut pac.RESETS,
+        rtc::DateTime {
+            year: 0,
+            month: 1,
+            day: 1,
+            day_of_week: rtc::DayOfWeek::Monday,
+            hour: 0,
+            minute: 0,
+            second: 0,
+        },
+    )
+    .unwrap();
+
+    // Trigger the IRQ every time a minute starts.
+    rtc.schedule_alarm(rtc::DateTimeFilter::default().second(0));
+    // Let the alarm trigger an interrupt in the NVIC.
+    rtc.enable_interrupt();
+
+    // Give away our rtc by moving them into the `GLOBAL_SHARED` variable.
+    // We won't need to access it in the main thread again
+    critical_section::with(|cs| {
+        GLOBAL_SHARED.borrow(cs).replace(Some(rtc));
+    });
+
+    // Let the core enter deep-sleep while waiting on wfi
+    core.SCB.set_sleepdeep();
+
+    // Unmask the RTC IRQ so that the NVIC interrupt controller
+    // will jump to the interrupt function when the interrupt occurs.
+    // We do this last so that the interrupt can't go off while
+    // it is in the middle of being configured
+    unsafe {
+        pac::NVIC::unmask(pac::Interrupt::RTC_IRQ);
+    }
+
+    cortex_m::asm::delay(XTAL_FREQ_HZ.to_Hz() / 2);
+    loop {
+        // interrupts handle everything else in this example.
```

Comment:
This comment is a little bit misleading because in this example, the actual work (toggling the LED) is not done from the interrupt handler.